tokio/src/io/async_fd.rs

```diff
@@ -817,7 +817,9 @@ impl<T: AsRawFd> Drop for AsyncFd<T> {
 impl<'a, Inner: AsRawFd> AsyncFdReadyGuard<'a, Inner> {
     /// Indicates to tokio that the file descriptor is no longer ready. All
     /// internal readiness flags will be cleared, and tokio will wait for the
-    /// next edge-triggered readiness notification from the OS.
+    /// next readiness notification after the last operation from the OS. This
+    /// implies that if a readiness notification occurs following the last operation
+    /// but prior to invoking `clear_ready`, it will not be cleared.
```

Comment:
Instead of modifying the current paragraph, I think it would be more clear to add a new paragraph at the end.